druid-shell/src/platform/x11/window.rs

```diff
@@ -289,20 +274,13 @@ struct WindowState {
 }
 
 impl Window {
-    fn connect(&self, handle: WindowHandle) -> Result<(), Error> {
-        match self.handler.try_borrow_mut() {
-            Ok(mut handler) => {
-                let size = self.size()?;
-                handler.connect(&handle.into());
-                handler.scale(Scale::default());
-                handler.size(size);
-                Ok(())
-            }
-            Err(err) => Err(Error::BorrowError(format!(
-                "Window::connect handler: {}",
-                err
-            ))),
-        }
+    fn connect(&self, handle: WindowHandle) -> Result<(), anyhow::Error> {
+        let mut handler = self.handler.try_borrow_mut().context("Window::connect")?;
```

Comment:
The context should contain the word *handler* like the old error so we know which variable this errored on.

Reply:
I've made these even a little more compact with some macros that include not only the variable that caused the error, but also the source file and line where it happened.